Add Android record keycode mapping, and send the key event to Kabuki

zhongqiliang opened the pull request:
b/272048725

I verified it using adb command to send the keycode.
We need to verify the TV remote record button triggers the keycode AKEYCODE_MEDIA_RECORD(130)

y4vor reviewed (commented):
You need to also update the Cobalt DOM keycoces. Those mirror SbKey.
https://github.com/youtube/cobalt/blob/main/cobalt/dom/keycode.h.

Also let's map to "MediaRecord"
https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values#multimedia_keys

https://github.com/youtube/cobalt/blob/main/cobalt/dom/keyboard_event.cc#L338

zhongqiliang commented:
Added the mapping for DOM keycode. 
My understanding is it will be used in the virtual keyboard in the html file, this is not tested. Added unit test to cover it.
